libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -87,6 +107,32 @@ void StructuralSimplifier::simplify(std::vector<yul::Statement>& _statements)
 					return s;
 				}
 			}
+			else if (auto const constExprVal = hasLiteralValue(*_switchStmt.expression))
+			{
+				Block* matchingCaseBlock = nullptr;
+				Case* defaultCase = nullptr;
+
+				for (auto& _case: _switchStmt.cases)
+				{
+					if (_case.value && valueOfNumberLiteral(*_case.value) == constExprVal)
```

Comment:
I think it would be safer to use an explicit `*constExprVal` here (especially since the actual type of `constExprVal` is hidden by the `auto` above ( ;-) ).

Reply:
Hmm I'd rather don't use the `auto` in that case. Also okay?